### PR TITLE
Add log() in stream-error.md

### LINKED ISSUE #23722: Add log() as an example on handling errors page in Akka Streams
As otherwise people often say "akka swallows exceptions" http://doc.akka.io/docs/akka/2.5.4/scala/stream/stream-error.html

### PR DESCRIPTION
**Pull request by richardimaoka**
Closes #23722 

paradox pages generated in my firebase demo site:

https://richard-akka-doc.firebaseapp.com/stream/stream-error.html#logging-errors
https://richard-akka-doc.firebaseapp.com/stream/stream-cookbook.html#logging-in-stream

### CODE REVIEW

File: akka-docs/src/main/paradox/stream/stream-cookbook.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ If you need a quick reference of the available processing stages used in the rec
 In this collection we show simple recipes that involve linear flows. The recipes in this section are rather
 general, more targeted recipes are available as separate sections (@ref:[Buffers and working with rate](stream-rate.md), @ref:[Working with streaming IO](stream-io.md)).
 
-### Logging elements of a stream
+### Logging in stream
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/stream/stream-cookbook.md]: Renaming the section is safe - confirmed no other place referred to this by `@ref:[***](stream-cookbook.md#loggign-elements-of-a-stream)`.

2018-02-11_15h21_44

File: akka-docs/src/test/java/jdocs/stream/javadsl/cookbook/RecipeLoggingElements.java [hunk 2]
```diff
@@ -68,14 +68,14 @@ public void workWithLog() throws Exception {
...
-        final int onFailure = Logging.ErrorLevel();
-
         //#log-custom
         // customise log levels
         mySource.log("before-map")
-          .withAttributes(Attributes.createLogLevels(onElement, onFinish, onFailure))
+          .withAttributes(Attributes.createLogLevels(
+            Logging.WarningLevel(), //onElement
+            Logging.InfoLevel(),    //onFinish
+            Logging.DebugLevel()    //onFailure
```
Comment (richardimaoka) [inline akka-docs/src/test/java/jdocs/stream/javadsl/cookbook/RecipeLoggingElements.java]: Moving these inside the visible code snippet, for users to easily find how to configure levels by these `Logging` factory methods.

File: akka-docs/src/test/scala/docs/stream/cookbook/RecipeLoggingElements.scala [hunk 3]
```diff
@@ -30,7 +30,13 @@ class RecipeLoggingElements extends RecipeSpec {
       //#log-custom
       // customise log levels
       mySource.log("before-map")
-        .withAttributes(Attributes.logLevels(onElement = Logging.WarningLevel))
+        .withAttributes(
+          Attributes.logLevels(
+            onElement = Logging.WarningLevel,
+            onFinish = Logging.InfoLevel,
+            onFailure = Logging.DebugLevel
```
Comment (richardimaoka) [inline akka-docs/src/test/scala/docs/stream/cookbook/RecipeLoggingElements.scala]: To show that users can configure different log level for each of element, finish and failure.

**Comment by richardimaoka** [comment]
ready for review now.

File: akka-docs/src/main/paradox/stream/stream-cookbook.md (hunk 1 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/stream/stream-cookbook.md]: New name sounds a bit weird though, perhaps making it plural would make it better: `Logging in streams`

File: akka-docs/src/test/java/jdocs/stream/javadsl/cookbook/RecipeLoggingElements.java (hunk 2 above)
Comment (johanandren) [reply akka-docs/src/test/java/jdocs/stream/javadsl/cookbook/RecipeLoggingElements.java]: 👍

File: akka-docs/src/test/scala/docs/stream/cookbook/RecipeLoggingElements.scala (hunk 3 above)
Comment (johanandren) [reply akka-docs/src/test/scala/docs/stream/cookbook/RecipeLoggingElements.scala]: 👍

**Review by johanandren — APPROVED** [review]
LGTM with the title changed

File: akka-docs/src/main/paradox/stream/stream-cookbook.md (hunk 1 above)
Comment (richardimaoka) [reply akka-docs/src/main/paradox/stream/stream-cookbook.md]: Thanks for the feedback. Added a new commit to fix this.

**Review by chbatey — APPROVED** [review]

**Pull request merged**